hera_qm/xrfi.py

```diff
@@ -1090,6 +1090,11 @@ def calculate_metric(uv, algorithm, cal_mode='gain', correlations='both', run_ch
         is raised.
 
     """
+    if issubclass(uv.__class__, (UVData)):
+        if correlations == 'auto':
+            uv = uv.select(bls=[(ant1, ant2) for ant1, ant2, in zip(uv.ant_1_array, uv.ant_2_array) if ant1 == ant2], inplace=False)
+        if correlations == 'cross':
+            uv = uv.select(bls=[(ant1, ant2) for ant1, ant2, in zip(uv.ant_1_array, uv.ant_2_array) if ant1 != ant2], inplace=False)
```

Comment:
I think you can do this select with `ant_str` instead of `bls` to streamline this code a bit.

Reply:
good point. Done this.